pkg/controller/account/account_controller.go

```diff
@@ -558,105 +558,11 @@ func matchSubstring(roleID, role string) (bool, error) {
 	return matched, err
 }
 
-// Returns true if account CR is Failed
-func accountIsFailed(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.State == AccountFailed
-}
-
-// Returns true of there is no state set
-func accountHasState(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.State != ""
-}
-
-// Returns true of there is no Status.SupportCaseID set
-func accountHasSupportCaseID(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.SupportCaseID != ""
-}
-
-func accountIsPendingVerification(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.State == AccountPendingVerification
-}
-
-func accountIsReady(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.State == AccountReady
-}
-
-func accountIsCreating(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Status.State == AccountCreating
-}
-
-func accountHasClaimLink(currentAcctInstance *awsv1alpha1.Account) bool {
-	return currentAcctInstance.Spec.ClaimLink != ""
-}
-
 func accountCreatingTooLong(currentAcctInstance *awsv1alpha1.Account) bool {
-	return accountIsCreating(currentAcctInstance) &&
+	return currentAcctInstance.IsCreating() &&
 		time.Now().Sub(currentAcctInstance.GetCreationTimestamp().Time) > createPendTime
 }
```

Comment:
Why not this one?

Reply:
Specifically because `createPendTime` is defined here in the controller.

I probably _could_ modify the function to take a duration and then compare against that, so then we don't have to define the pending time within the account CR (which I don't think would be a good place for it).  I might just do that 🤔